### ISSUE TITLE
TracerWarning when using the DeformConvPack

### ISSUE THREAD

**Issue by jonas154** [open-mmlab/mmdetection#2247]
Hi everyone, 

I'm using your implementation of the deformable convolution in one of my projects. However, I'm always facing TracerWarning's - are they "normal"?

> /snap/pycharm-professional/183/plugins/python/helpers/pydev/_pydevd_bundle/pydevd_xml.py:371: TracerWarning: Converting a tensor to a Python index might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  xml_shape = ' shape="%s"' % make_valid_xml_value("%s" % str(len(v)))
/mmdetection/mmdet/ops/dcn/deform_conv.py:107: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  if not all(map(lambda s: s > 0, output_size)):
/mmdetection/mmdet/ops/dcn/deform_conv.py:48: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  cur_im2col_step = min(ctx.im2col_step, input.shape[0])
/mmdetection/mmdet/ops/dcn/deform_conv.py:50: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  cur_im2col_step) == 0, 'im2col step must divide batchsize'
/mmdetection/mmdet/ops/dcn/deform_conv.py:56: TracerWarning: Converting a tensor to a Python integer might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  cur_im2col_step)

Thats the way, how I use the layer:

`    def __init__(self):
        super(TestNetwork, self).__init__()

        self.conv1 = DeformConvPack(1, 16, kernel_size=3, padding=1, dilation=1, stride=1)
        self.batch_norm_1 = nn.BatchNorm2d(16)
        self.relu1 = nn.ReLU(inplace=True)

        self.max_pool1 = nn.MaxPool2d(2, 2)

        self.fc1 = nn.Linear(16 * 64 * 16, 200)
        self.reluFC = nn.ReLU(inplace=True)
        self.fc2 = nn.Linear(200, 64)

    def forward(self, x):
        out = self.conv1(x)

        out = self.batch_norm_1(out)
        out = self.relu1(out)
        out = self.max_pool1(out)
        rois = torch.randn(20, 5).cuda()
        #     rois = torch.randn(2, 18, 128, 32).cuda()
        # out = self.dpooling(out, rois)

        out = torch.flatten(out, 1)
        out = self.fc1(out)
        out = self.reluFC(out)
        out = self.fc2(out)

        return out`

Thanks and best regards
Jonas

**Comment by xvjiarui** [comment]
Hi @jonas154 
Sorry for the late reply. May I know the details about your environment?

**Comment by jonas154** [comment]
Good Morning @xvjiarui, 
year of cause:

Python: 3.7.6
Pytorch: 1.4
Sys Platform: Linux / Ubuntu 16.04
CUDA Version: 10.2
Nvidia Driver Version: 440.33.01 

And I'm on the master branch of the open-mmlab, at git commit 050614a9b8528de4f8e453aa42d9265206738eb1

However, what came to me today in mind is that I compiled the Repo under the git commit 7473533c706ef8f6ef4132f55a75cc47dd57d8e3. Should I recompile everything?

Thanks and best regards
Jonas

**Comment by xvjiarui** [comment]
Sorry for the late reply.
There is no need to do so. I noticed that it may due to that DCN is a customize operator. But it won't affect the performance. So please feel free to ignore it.

**Comment by xvjiarui** [comment]
Hi @jonas154 
You may also try if this warning exists in MMDet v2.0. 
Please feel free to reopen this issue.
